### PR TITLE
Game creation & Player joining

### PR DESCRIPTION
**Pull request by DrakenLine**

### CODE REVIEW

**Comment by DrakenLine** [comment]
Task linked: [CU-860rhrz8x Handle "player joined" event](https://app.clickup.com/t/860rhrz8x)

File: lib/Boat/boat.dart [hunk 1]
```diff
@@ -18,10 +21,10 @@ class Boat {
 
   factory Boat.fromApiResponse(Map<String, dynamic> data) {
     return Boat(
-      name: data['name'],
-      beam: data['beam'],
-      lengthOverall: data['lengthOverall'],
-      boatDirection: data['boatDirection'],
+      name: getBoatName(data['name'] ?? ""),
+      beam: (data['beam'] as int).toDouble(),
+      lengthOverall: (data['lengthOverall'] as int).toDouble(),
```
Comment (AlexandreMichelix) [inline lib/Boat/boat.dart:26]: Can you avoid cast int ?

File: lib/Boat/constants.dart
Comment (AlexandreMichelix) [inline lib/Boat/constants.dart:1]: sort enum please

File: lib/Boat/helpers.dart
Comment (AlexandreMichelix) [inline lib/Boat/helpers.dart:1]: sort switch ?

**Review by AlexandreMichelix — APPROVED** [review]

**Review by AlexandreMichelix — APPROVED** [review]

**Pull request merged**